### PR TITLE
Add Javadoc autogeneration that works on Jitpack

### LINKED ISSUE #60: Publish javadoc on jitpack.io
Jitpack automatically publishes javadoc for every release (even for `develop`) if your project generates a `javadoc.jar`. See https://jitpack.io/docs/#features

Material Motion gets this for free due to our one-module-per-repo setup
https://jitpack.io/com/github/material-motion/material-motion-runtime-android/4.0.0/javadoc/
https://jitpack.io/com/github/material-motion/material-motion-family-rebound-android/1.0.0/javadoc/

Since you have multiple modules, you'll probably have to do something extra to get it to work https://jitpack.io/com/github/material-foundation/material-remixer-android/0.5.0/

### PR DESCRIPTION
**Pull request by miguelandres**
Generates javadoc that includes the whole scope of remixer_core, remixer_annotation, remixer_ui and remixer_storage, merging all of them into one source set.

Hacks had to be made to accommodate the fact that some are java libraries and some are android libraries.

Added a link to the develop branch snapshot's javadoc to the Readme. this link won't work until this diff is merged but you can preview it [on the javadoc branch snapshot](https://jitpack.io/com/github/material-foundation/material-remixer-android/remixer/javadoc-SNAPSHOT/javadoc/index.html).

Fixes #60

### CODE REVIEW

**Comment by chuga** [comment]
👏

File: remixer/build.gradle [hunk 1]
```diff
@@ -51,3 +51,42 @@ dependencies {
...
+]
+
+// Make this project depend on complete evaluation of all the javadoc-generating projects.
+// This lets us use the ext variables generated in afterEvaluate.
+javadocProjects.each {
+  evaluationDependsOn(it)
+}
+
+// Task that generates javadoc scoped across all subprojects in javadocProjects.
+task alljavadoc(type: Javadoc) {
```
Comment (pingpongboss) [inline remixer/build.gradle]: "allJavadoc" would follow naming conventions

File: build.gradle [hunk 2]
```diff
@@ -45,6 +45,29 @@ subprojects {
     configFile = rootProject.file('checkstyle.xml')
     toolVersion = '7.1'
   }
+
+  ext {
+    sourceSetForJavadoc = null
+    classpathForJavadoc = null
+  }
```
Comment (pingpongboss) [inline build.gradle:52]: is this necessary?

**Review by pingpongboss — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (miguelandres) [reply build.gradle:52]: It complains if it's not declared otherwise.

File: remixer/build.gradle (hunk 1 above)
Comment (miguelandres) [reply remixer/build.gradle]: Done.

**Pull request merged**